### DISCUSSION TITLE
Diving Suit Locker Signal Out

### DISCUSSION THREAD

**Discussion by SophieBaybey** [FakeFishGames/Barotrauma#8854] [Ideas and suggestions]
Pretty simple request, add a signal output for when a diving suit locker has a suit inside of it. I often find myself rampantly looking for suits in the sub when an emergency hits, only to find they've all been taken. I'd love to be able to wire up a little display to show how many lockers actively have suits in them.

**Comment by mLuby** [comment]
A Diving Suit Locker (and Oxygen Tank Shelf) could have a `quantity_out` pin for how many items it currently contains: usually 0-1 for lockers and 0-3 for shelves.

Separately but related, they should _only_ draw power when recharging the contained items, not all the time like they do now.

Putting those two changes together, circuits could then tell whether it's empty, charging, or full.


quantity_out | power_in | meaning
:-- | :-- | :--
0 | 0 | Empty
3 | 0 | Charged (3 tank/suits)
3 | 30kW | Charging (3 tanks/suits)
0 | 30kW | Malfunction (shouldn't be possible)

**Reply by Oukido** [reply]
good idea, they should also do that for charging docks
